src/codeu/chat/server/Controller.java

```diff
@@ -207,6 +209,47 @@ public Time statusUpdate() {
 	  return null; 
   }
 
-  
+  public Uuid changeAccessControl(Uuid userID, Uuid convo, String username, UserType type)
+  {
+    // This function will update the UserType of the User with the User Name username
+    // If they have the right to. If not, it will not update and simply return null
+    if(model.conversationById().first(convo).getAccessOf(userID)== UserType.CREATOR)
+    {
+      User toUpdate = model.userByText().first(username);
+      if (toUpdate == null)
+      {
+        return null;
+      }
+      if(type == UserType.OWNER || type == UserType.MEMBER) {
+        model.conversationById().first(convo).setAccessOf(model.userByText().first(username).id, type);
+        return model.userByText().first(username).id;
+      }
+      else
+      {
+        return null;
+      }
+    }
+
+    if(model.conversationById().first(convo).getAccessOf(userID)== UserType.OWNER)
+    {
+      User toUpdate = model.userByText().first(username);
+      if (toUpdate == null)
+      {
+        return null;
+      }
+      if(type == UserType.MEMBER)
+      {
+        model.conversationById().first(convo).setAccessOf(model.userByText().first(username).id,type);
+        return model.userByText().first(username).id;
+      }
+      else
```

Comment:
You probably don't need this `else` block since you return `null` right after it anyway?